07_modules_libs/homework/request.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/go-chi/chi/v5"
+	"io/ioutil"
+	"net/http"
+	"time"
+)
+
+const apiKey = "2NVMJS308BND1F0Y"
+
+func getStockInfo(w http.ResponseWriter, r *http.Request) {
+
+	stockName := chi.URLParam(r, "Ticker")
+
+	url := fmt.Sprintf("https://www.alphavantage.co/query?function=TIME_SERIES_DAILY_ADJUSTED&symbol=%s&apikey=%s", stockName, apiKey)
+
+	response, error := http.Get(url)
+
+	if error != nil {
+		writeResponse(w, http.StatusBadRequest, error.Error())
+		return
+	}
+
+	body, error2 := ioutil.ReadAll(response.Body)
```

Comment:
You should handle error right after it was returned. And no need to call variable `error2` or `err2`